### PR TITLE
network: fix vm can be deployed on L2 network of other accounts

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR fixes a bug with integration test test_storage_policy.py
https://github.com/apache/cloudstack/blob/4.16.0.0/test/integration/smoke/test_storage_policy.py#L191-L208

in the test, root admin deploys a vm for another account on root admin's L2 network, it should fail but test passes.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [x] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan test

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Review by GabrielBrascher — APPROVED** [review]
Thanks, @weizhouapache.
Code LGTM!

**Comment by weizhouapache** [comment]
this PR needs more love.
related to #2420

File: server/src/main/java/com/cloud/network/NetworkModelImpl.java [hunk 1]
```diff
@@ -1659,8 +1659,7 @@ public void checkNetworkPermissions(Account owner, Network network) {
             throw new CloudRuntimeException("cannot check permissions on (Network) <null>");
         }
         // Perform account permission check
-        if ((network.getGuestType() != GuestType.Shared && network.getGuestType() != GuestType.L2) ||
-                (network.getGuestType() == GuestType.Shared && network.getAclType() == ACLType.Account)) {
+        if (network.getGuestType() != GuestType.Shared || network.getAclType() == ACLType.Account) {
```
Comment (sureshanaparti) [inline server/src/main/java/com/cloud/network/NetworkModelImpl.java:1662]: @weizhouapache check the changes introduced at https://github.com/apache/cloudstack/pull/2420, for L2 network.

**Comment by weizhouapache** [comment]
@nvazquez 
the issue fixed by #2420  is not a real issue in my opinion.
#3158 has reverted part of #2420, this PR reverts rest of #2420.
could you please review it ?

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test

**Review by rohityadavcloud — APPROVED** [review]

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@weizhouapache can you check unit test error
```
[ERROR] addUserAccountFail(org.apache.cloudstack.cloudian.CloudianClientTest)  Time elapsed: 9.42 s  <<< ERROR!
org.apache.cloudstack.api.ServerApiException: Operation timed out, please try again.
	at org.apache.cloudstack.cloudian.CloudianClientTest.addUserAccountFail(CloudianClientTest.java:145)
```

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
> 

@sureshanaparti it is not related to this PR. let me rebuild the package

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan test

**Pull request merged**